File: src/Http.API/ClientApp/src/app/share/admin/services/entity-model.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { EntityModelFilterDto } from '../models/entity-model/entity-model-filter-dto.model';
import { EntityModelAddDto } from '../models/entity-model/entity-model-add-dto.model';
import { EntityModelUpdateDto } from '../models/entity-model/entity-model-update-dto.model';
import { EntityModelItemDtoPageList } from '../models/entity-model/entity-model-item-dto-page-list.model';
import { EntityModel } from '../models/entity-model/entity-model.model';

/**
 * 实体模型类
 */
@Injectable({ providedIn: 'root' })
export class EntityModelService extends BaseService {
  /**
   * 筛选
   * @param data EntityModelFilterDto
   */
  filter(data: EntityModelFilterDto): Observable<EntityModelItemDtoPageList> {
    const url = `/api/admin/EntityModel/filter`;
    return this.request<EntityModelItemDtoPageList>('post', url, data);
  }

  /**
   * 新增
   * @param data EntityModelAddDto
   */
  add(data: EntityModelAddDto): Observable<EntityModel> {
    const url = `/api/admin/EntityModel`;
    return this.request<EntityModel>('post', url, data);
  }

  /**
   * 更新
   * @param id 
   * @param data EntityModelUpdateDto
   */
  update(id: string, data: EntityModelUpdateDto): Observable<EntityModel> {
    const url = `/api/admin/EntityModel/${id}`;
    return this.request<EntityModel>('put', url, data);
  }

  /**
   * 详情
   * @param id 
   */
  getDetail(id: string): Observable<EntityModel> {
    const url = `/api/admin/EntityModel/${id}`;
    return this.request<EntityModel>('get', url);
  }

  /**
   * ⚠删除
   * @param id 
   */
  delete(id: string): Observable<EntityModel> {
    const url = `/api/admin/EntityModel/${id}`;
    return this.request<EntityModel>('delete', url);
  }

}

File: src/Sdks/Dusi.Manage.Client/Services/EntityModelService.cs
using Share.Models.EntityModelDtos;
namespace Dusi.Manage.Client.Services;
/// <summary>
/// 实体模型类
/// </summary>
public class EntityModelService : BaseService
{
    public EntityModelService(HttpClient httpClient) : base(httpClient)
    {

    }
    /// <summary>
    /// 筛选
    /// </summary>
    /// <param name="data">EntityModelFilterDto</param>
    /// <returns></returns>
    public async Task<PageList<EntityModelItemDto>?> FilterAsync(EntityModelFilterDto data)
    {
        string url = $"/api/admin/EntityModel/filter";
        return await PostJsonAsync<PageList<EntityModelItemDto>?>(url, data);
    }

    /// <summary>
    /// 新增
    /// </summary>
    /// <param name="data">EntityModelAddDto</param>
    /// <returns></returns>
    public async Task<EntityModel?> AddAsync(EntityModelAddDto data)
    {
        string url = $"/api/admin/EntityModel";
        return await PostJsonAsync<EntityModel?>(url, data);
    }

    /// <summary>
    /// 更新
    /// </summary>
    /// <param name="id"> </param>
    /// <param name="data">EntityModelUpdateDto</param>
    /// <returns></returns>
    public async Task<EntityModel?> UpdateAsync(string id, EntityModelUpdateDto data)
    {
        string url = $"/api/admin/EntityModel/{id}";
        return await PutJsonAsync<EntityModel?>(url, data);
    }

    /// <summary>
    /// 详情
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<EntityModel?> GetDetailAsync(string id)
    {
        string url = $"/api/admin/EntityModel/{id}";
        return await GetJsonAsync<EntityModel?>(url);
    }

    /// <summary>
    /// ⚠删除
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<EntityModel?> DeleteAsync(string id)
    {
        string url = $"/api/admin/EntityModel/{id}";
        return await DeleteJsonAsync<EntityModel?>(url);
    }

}